### PR TITLE
docs: migrate docusaurus deployment from surge to github pages

### LINKED ISSUE #126: [docs] Migrate docusaurus deployment from surge to gh-pages
### Is your feature request related to a problem? Please describe.
It will be better to have the docs on the github-pages.

### Describe the solution you'd like
Deploy the docs on gh-pages.
Take reference from this: https://github.com/anitab-org/mentorship-backend/pull/841/commits/4cc78ae98b058a2eb2ad33adf1018e19e3a70e54

### PR DESCRIPTION
**Pull request by Harshita-Kanal**
### Description

Migrate docusaurus deployment from surge to github pages.

Fixes #126 

### Type of Change:

- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          git config --global user.name "gh-actions"
+          cd docs/
+          if [ -e yarn.lock ]; then
+          yarn install --frozen-lockfile
+          elif [ -e package-lock.json ]; then
+          npm ci
+          else
+          npm i
+          fi
+          npm run deploy
```
Comment (codesankalp) [inline .github/workflows/docs.yml]: add a new line at the end of file.

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Open Source Programs (OSP) is an application that simplifies the processing and
 
 ## Documentation
 
-Documentation for the project is hosted [here](https://osp-web-docs.surge.sh/). We use `Docusaurus` for maintaining the documentation of the project.
+Documentation for the project is hosted [here](https://anitab-org.github.io/open-source-programs/). We use `Docusaurus` for maintaining the documentation of the project.
```
Comment (codesankalp) [inline README.md]: Please use anitab-forms-web

**Review by codesankalp — CHANGES_REQUESTED** [review]

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
 module.exports = {
   title: 'Open Source Programs Web Docs',
   tagline: 'Documentation for Open Source Programs Web',
-  url: 'https://osp-web-docs.surge.sh',
-  baseUrl: '/',
+  url: 'https://anitab-org.github.io',
+  baseUrl: '/open-source-programs/',
```
Comment (Aaishpra) [inline docs/docusaurus.config.js]:
```suggestion
  baseUrl: '/anitab-forms-web/',
```

**Comment by Harshita-Kanal** [comment]
Added the changes

**Review by Aaishpra — DISMISSED** [review]
LGTM 💯

**Review by Amulya-coder — DISMISSED** [review]
Looks good @Harshita-Kanal, could you please fix the conflicts.

**Comment by Harshita-Kanal** [comment]
Fixed the conflict

**Review by Amulya-coder — APPROVED** [review]

**Review by vj-codes — APPROVED** [review]
LGTM 🎉

**Comment by vj-codes** [comment]
@Aaishpra can you review this again please?

**Review by Aaishpra — APPROVED** [review]
Great work here 👍

**Review by isabelcosta — APPROVED** [review]
Great work @Harshita-Kanal 🙌🏾  Thank you for contributing!

**Comment by isabelcosta** [comment]
I just set this the GH_PAGES_DEPLOY key, I hope it works now 🙏🏾

**Pull request merged**